libraries/AP_InertialSensor/AP_InertialSensor_Invensensev3.cpp

```diff
@@ -609,13 +675,47 @@ void AP_InertialSensor_Invensensev3::set_filter_and_scaling_icm42670(void)
     register_write(INV3REG_70_ACCEL_CONFIG1, 0x01);
 }
 
+/*
+  set the filter frequencies and scaling for the ICM-456xy
+ */
+void AP_InertialSensor_Invensensev3::set_filter_and_scaling_icm456xy(void)
+{
+    backend_rate_hz = 1600;
+
+    // Disable FIFO first
+    register_write(INV3REG_456_FIFO_CONFIG3, 0x00);
+    register_write(INV3REG_456_FIFO_CONFIG0, 0x07);
+
+
+    // setup gyro for 1.6kHz, 4000dps range
+    register_write(INV3REG_456_GYRO_CONFIG0, 0x05);
+
+    // setup accel for 1.6kHz, 32g range
+    register_write(INV3REG_456_ACCEL_CONFIG0, 0x05);
+
+    // enable timestamps on FIFO data
+    // SMC_CONTROL_0
+    uint8_t reg = register_read_bank_icm456xy(INV3BANK_456_IPREG_TOP1_ADDR, 0x58);
+    register_write_bank_icm456xy(INV3BANK_456_IPREG_TOP1_ADDR, 0x58, reg | 0x01);
+
+    // enable FIFO for each sensor
+    register_write(INV3REG_456_FIFO_CONFIG3, 0x06, true);
+
+    // FIFO stop-on-full, disable bypass and 2K FIFO
+    register_write(INV3REG_456_FIFO_CONFIG0, 0x87, true);
+
+    // enable FIFO
+    register_write(INV3REG_456_FIFO_CONFIG3, 0x07, true);
+}
+
 /*
   check whoami for sensor type
  */
 bool AP_InertialSensor_Invensensev3::check_whoami(void)
 {
     uint8_t whoami = register_read(INV3REG_WHOAMI);
 
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Inv3: Whoami 0x%02x", whoami);
```

Comment:
This seems pretty cryptic for an info message - maybe use debug/hal.console->printf()?